apiserver/logsink/filewriter.go

```diff
@@ -29,11 +32,14 @@ func NewFileWriter(logPath string) (io.WriteCloser, error) {
 // primeLogFile ensures the logsink log file is created with the
 // correct mode and ownership.
 func primeLogFile(path string) error {
-	f, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY, 0600)
+	permissions := os.FileMode(0640)
+	f, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY, permissions)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	f.Close()
-	err = utils.ChownPath(path, "syslog")
-	return errors.Trace(err)
+	if err := f.Close(); err != nil {
+		return errors.Trace(err)
+	}
+	wantedOwner, wantedGroup := systemd.SyslogUserGroup()
```

Comment:
This line seems out of place.

Since we aren't calling a generic function to set the Owner and Group permissions, but one in `auditlog`, it seems that the `systemd.SyslogUserGroup` should be inside that function.

I think perhaps we should split out the permissions as well. They are useful for the upgrade step, but likely that we don't need them elsewhere. We know the permissions above.

This would mean that the function call just becomes
```go
auditlog.SetOwnerGroupLogPermissions(path)
```